planning/behavior_path_planner/include/behavior_path_planner/utils/start_goal_planner_common/utils.hpp

```diff
@@ -0,0 +1,76 @@
+// Copyright 2023 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BEHAVIOR_PATH_PLANNER__UTILS__START_GOAL_PLANNER_COMMON__UTILS_HPP_
+#define BEHAVIOR_PATH_PLANNER__UTILS__START_GOAL_PLANNER_COMMON__UTILS_HPP_
+
+#include "behavior_path_planner/data_manager.hpp"
+#include "behavior_path_planner/utils/goal_planner/goal_planner_parameters.hpp"
+#include "behavior_path_planner/utils/path_safety_checker/path_safety_checker_parameters.hpp"
+#include "behavior_path_planner/utils/start_planner/pull_out_path.hpp"
+#include "behavior_path_planner/utils/start_planner/start_planner_parameters.hpp"
+
+#include <motion_utils/distance/distance.hpp>
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+namespace behavior_path_planner::utils::start_goal_planner_common
+{
+
+using behavior_path_planner::StartPlannerParameters;
+using behavior_path_planner::utils::path_safety_checker::EgoPredictedPathParams;
+using behavior_path_planner::utils::path_safety_checker::ObjectsFilteringParams;
+using behavior_path_planner::utils::path_safety_checker::SafetyCheckParams;
+
+boost::optional<double> calcFeasibleDecelDistance(
+  std::shared_ptr<const PlannerData> planner_data, const double acc_lim, const double jerk_lim,
+  const double target_velocity);
```

Comment:
remove from goal planner module and use this for it.

Reply:
Removed from goal planne module in https://github.com/autowarefoundation/autoware.universe/pull/4804/commits/2b45e9ba1dfa1af9488d634dc540ab915ad7385d